androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/components/playerOverlay/VolumeControl.kt

```diff
@@ -0,0 +1,88 @@
+package com.profusion.androidenhancedvideoplayer.components.playerOverlay
+
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.height
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import com.profusion.androidenhancedvideoplayer.styling.Dimensions
+
+private const val MIN_VOLUME_VALUE = 0
+
+enum class VolumeIconState {
+    HIGH,
+    LOW,
+    OFF
+}
+
+fun mapVolumeRangeToState(volume: Int, maxVolumeValue: Int): VolumeIconState {
+    val halfVolume = maxVolumeValue / 2
+    val lowRange = MIN_VOLUME_VALUE + 1..halfVolume
+    val highRange = halfVolume + 1..maxVolumeValue
+    return when (volume) {
+        in lowRange -> VolumeIconState.LOW
+        in highRange -> VolumeIconState.HIGH
+        else -> VolumeIconState.OFF
+    }
+}
+
+fun mapVolumeIconStateToIcon(
+    state: VolumeIconState,
+    customization: ControlsCustomization
+): @Composable () -> Unit {
+    return when (state) {
+        VolumeIconState.LOW -> customization.volumeLowIconContent
+        VolumeIconState.OFF -> customization.volumeOffIconContent
+        VolumeIconState.HIGH -> customization.volumeHighIconContent
+    }
+}
+
+@Composable
+fun VolumeControl(
+    modifier: Modifier = Modifier,
+    shouldEnableVolumeControl: Boolean,
+    isFullScreen: Boolean,
+    volume: () -> Int,
+    maxVolumeValue: () -> Int,
+    setDeviceVolume: (Int) -> Unit,
+    customization: ControlsCustomization,
+    volumeMutableInteractionSource: MutableInteractionSource
+) {
+    if (isFullScreen && shouldEnableVolumeControl) {
```

Comment:
And this component doesn't need to know if is fullscreen or not, let its parent decide if the fullscreen will impact the visibility through the `shouldEnableVolumeControl` method